### ISSUE TITLE
"build.command" failed (Eleventy requires Node 10)

### ISSUE THREAD

**Issue by smth** [11ty/eleventy#1931]
I've just updated a site to Eleventy 0.12.1 (from 0.7.1). Everything seems fine locally (WIndows 10 / WSL 1), but builds are now failing on Netlify. I'm getting `"build.command" failed` preceded by `Eleventy requires Node 10. You will need to upgrade Node to use Eleventy!`. Edit: Build log says it's using Node v8.17.0

![image](https://user-images.githubusercontent.com/1375708/129478781-fea0f60d-3bcf-46d5-ba7e-806c848c2601.png)

I've got the [netlify-plugin-subfont](https://github.com/Munter/netlify-plugin-subfont) Netlify plugin in there, which is probably noteworthy, but the error doesn't suggest to me that the failure is there.

**Comment by zachleat** [comment]
Your netlify build log should tell you what version of Node you're running, can you include that?

**Comment by smth** [comment]
Yeah, Node v8.17.0.

**Comment by pdehaan** [comment]
Some possibly relevant links:

https://github.com/11ty/eleventy/issues/1701; Bump Node minimum to v10+

After some quick digging through <kbd>git blame</kbd> for the package.json `engines` key, it looks like these were some noteworthy commits where the minimum required Node version was changed:
https://github.com/11ty/eleventy/commit/cdb1191360a53ba7defc2b93b1ec2895d031c368 (8=>10)
https://github.com/11ty/eleventy/commit/57e28374ff54f325603b0ba2dc5ca97d001f8634 (10=>12)

And just because I already had a [script handy](https://github.com/pdehaan/npm-package-info) that inspects packages and historical versions, here's the release version numbers where the package.json `engines.node` keys was modified:

```
@11ty/eleventy v0.2.0 Node version: >=8.0.0
@11ty/eleventy v0.5.1 Node version: >=8
@11ty/eleventy v0.12.1 Node version: >=10
@11ty/eleventy v1.0.0-canary.11 Node version: >=12
```

**Comment by smth** [comment]
I just tried dropping the Eleventy version to 0.11, which as expected removed the line about Node version. The build still failed however, with the same `"build.command" failed ` error.

**Comment by zachleat** [comment]
Ah, sorry—I’m a bit confused. Eleventy 0.12 requires Node 10+

https://github.com/11ty/eleventy/releases/tag/v0.12.0 and on the home page: https://www.11ty.dev/

![image](https://user-images.githubusercontent.com/39355/129489505-e6408de7-7c69-46bc-8966-2a9c6fdb802f.png)

You can control the Netlify Node version via a couple of different methods: https://docs.netlify.com/configure-builds/manage-dependencies/#node-js-and-javascript

**Comment by zachleat** [comment]
TLDR: I would recommend using 0.12.1 but with an `.nvmrc` file with the contents: `14`

**Comment by smth** [comment]
Sorry for the confusion, it seems the Node version was a red herring. Either with Eleventy 0.11 and Node 8 or Eleventy 0.12 and Node 14 I still get the same `"build.command" failed` error.

**Comment by zachleat** [comment]
Oh oh, I think you want `npx @11ty/eleventy`. Running `eleventy` requires Global installation, which isn’t recommended on Netlify Some info about this on https://www.11ty.dev/docs/global-installation/

**Comment by smth** [comment]
Ah I think that's got us closer to the problem. I'm still getting the same error but with the addition of a new one :) Looks like it's trying to render stuff in `node_modules`? - `Having trouble rendering liquid (and markdown) template ./.netlify/plugins/node_modules/acorn/README.md`

```
Language does not exist: sh
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble rendering liquid (and markdown) template ./.netlify/plugins/node_modules/acorn/README.md
`TemplateContentRenderError` was thrown
> Cannot read property 'rest' of undefined
`TypeError` was thrown:
TypeError: Cannot read property 'rest' of undefined
        at Object.tokenize (/opt/build/repo/node_modules/prismjs/prism.js:658:23)
        at Object.highlight (/opt/build/repo/node_modules/prismjs/prism.js:628:19)
        at Object.module.exports [as highlight] (/opt/build/repo/node_modules/@11ty/eleventy-plugin-syntaxhighlight/src/markdownSyntaxHighlight.js:24:18)
        at Object.default_rules.fence (/opt/build/repo/node_modules/markdown-it/lib/renderer.js:50:27)
        at Renderer.render (/opt/build/repo/node_modules/markdown-it/lib/renderer.js:326:38)
        at MarkdownIt.render (/opt/build/repo/node_modules/markdown-it/lib/index.js:543:24)
        at /opt/build/repo/node_modules/@11ty/eleventy/src/Engines/Markdown.js:72:38
        at processTicksAndRejections (internal/process/task_queues.js:95:5)
        at async Template.render (/opt/build/repo/node_modules/@11ty/eleventy/src/TemplateContent.js:199:22)
        at async Template._getContent (/opt/build/repo/node_modules/@11ty/eleventy/src/Template.js:547:12)
```

**Comment by zachleat** [comment]
`.gitignore` your `node_modules`. I think Eleventy is processing those files (specifically in the markdown syntax highlighter)

**Comment by smth** [comment]
Do you mean `.eleventyignore`? `node_modules` is already git-ignored.

**Comment by zachleat** [comment]
Is this open source? can you provide a link to the repo? or invite me to a private repo

**Comment by smth** [comment]
I've just pushed it to GitHub and invited you @zachleat

**Comment by zachleat** [comment]
Did you fix it? Deploys fine for me both locally and on Netlify. If it’s still broken for you, try the steps here: https://docs.netlify.com/configure-builds/troubleshooting-tips/#build-cache

**Comment by smth** [comment]
That's as it was when it was failing. Clearing the build cache worked though :thumbsup: Thank you @zachleat :smiley:

**Comment by zachleat** [comment]
Awesome, thank you!

**Comment by zachleat** [comment]
Feel free to remove my permissions to that private repo 🙌🏻
